Added more info about Wasp using latest NodeJS LTS version.

Martinsos opened the pull request:
# Description

Provides motivation for Wasp requiring a specific Node version.

## Type of change

Please select the option(s) that is more relevant.

- [ ] Code cleanup
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

sodic commented on waspc/src/Wasp/Generator/Job/Process.hs:
```diff
@@ -167,6 +167,6 @@ makeNodeVersionMismatchMessage nodeVersion =
 waspNodeRequirementMessage =
   unwords
-    [ "Wasp requires node " ++ show C.nodeVersionBounds ++ ".",
+    [ "Wasp requires Node.js to be the latest LTS version, which is currently v" ++ show (SV.major C.nodeVersion) ++ ".",
```
Keep in mind that such wording calls for a quick update as soon as Node releases a new LTS version, which might be a good thing.

Still, what if we don't want to support the newest LTS right away? We can remove the line, but all existing installations of Wasp will technically be giving out wrong information.

For example, Node 18 becomes the new LTS version. We don't want to update it because it breaks something. Even if we change the wording here, all existing Wasp installations still say "Wasp requires Node.js to be the latest LTS version, which is currently v16.". This now becomes incorrect and can potentially give an impression that the project is out of date or poorly maintained.

sodic replied:
Also, I see you've swapped `^16.0.0` with `v16` probably because the first variant, although technically more informative, requires the users to understand semver syntax.

Still, saying just `v16` might leave them with the impression that their version must be 16.0.0, which is incorrect.

Perhaps the best option is to print something like `v16.x.x`. We can do this by changing `VersionBound`'s show method.

Martinsos replied:
Makes sense -> I will revert message to be what it was before, that is the easiest and most correct!

sodic commented on web/docs/getting-started.md:
```diff
@@ -16,6 +16,14 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+:::info
+
+Why this specific Node.js version and not a newer one?
+Because at the moment of the latest Wasp release, this specific Node.js version was the latest LTS ("long-term support") version of Node.js, which guarantees that it is stable, ready for general usage, it is actively maintained, and is recommended for use in production, unlike the newer versions which are still maturing.
+Check out https://nodejs.org/en/about/releases/ for more details.
```
I'd probably simplify this a bit:
Instead of:
> [..], which guarantees that it is stable, ready for general usage, it is actively maintained, and is recommended for use in production, unlike the newer versions which are still maturing.

Perhaps this:
> [..]. The latest LTS version guarantees stability and active maintenance, which is why the official Node.js team recommends it for usage in production.

sodic commented on web/docs/getting-started.md:
```diff
@@ -16,6 +16,14 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+:::info
+
+Why this specific Node.js version and not a newer one?
+Because at the moment of the latest Wasp release, this specific Node.js version was the latest LTS ("long-term support") version of Node.js, which guarantees that it is stable, ready for general usage, it is actively maintained, and is recommended for use in production, unlike the newer versions which are still maturing.
```
I'd probably repeat the specific version explicitly:
> ##### Why specifically Node.js 16.x.x and not a newer version?
>
> Because, at the moment of the latest Wasp release, Node.js `16.x.x` was the latest LTS ("long-term support") version of Node.js.

Martinsos commented:
Thanks for input @sodic ! Based on it I simplified the messaging and removed the mention of LTS from the CLI (but left it in docs, however better explained).

I will close this PR and instead move these doc changes to the PR that limits Wasp to node <=16.15.0 .
